source/adapters/cuda/enqueue.cpp

```diff
@@ -1756,3 +1756,33 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueWriteHostPipe(
 
   return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
 }
+
+UR_APIEXPORT ur_result_t UR_APICALL urEnqueueTimestampRecordingExp(
+    ur_queue_handle_t hQueue, bool blocking, uint32_t numEventsInWaitList,
+    const ur_event_handle_t *phEventWaitList, ur_event_handle_t *phEvent) {
+
+  ur_result_t Result = UR_RESULT_SUCCESS;
+  std::unique_ptr<ur_event_handle_t_> RetImplEvent{nullptr};
+  try {
+    ScopedContext Active(hQueue->getContext());
```

Comment:
```suggestion
    ScopedContext Active(hQueue->getDevice());
```
I think this might be necessary once https://github.com/intel/llvm/pull/10737 or similar is merged. CUcontext maps 1:1 with ur device but won't map 1:1 with ur context.

Reply:
ah sorry, my mistake I think that `getDevice()` is not yet implemented in the cuda adapter (only hip), so for that one I think the original code was correct.

Reply:
I see! It has been reverted. 😄